src/uk/gov/hmcts/contino/AzPublicDns.groovy

```diff
@@ -0,0 +1,57 @@
+package uk.gov.hmcts.contino
+
+import uk.gov.hmcts.contino.azure.Az
+
+class AzPublicDns {
+
+    def steps
+    def environment
+    def az
+    def environmentDnsConfigEntry
+    
+    AzPublicDns(steps,environment, environmentDnsConfigEntry) {
+      this.steps = steps
+      this.environment = environment
+      this.az = new Az(this.steps, this.steps.env.SUBSCRIPTION_NAME)
+      this.environmentDnsConfigEntry = environmentDnsConfigEntry
+    }
+
+   def getHostName(recordName) {
+     def zone = this.environmentDnsConfigEntry.zone
+
+     return "${recordName}.${zone}"
+   }
+
+    def registerDns(recordName, serviceIP) {
```

Comment:
this isn't registering dns it's checking if a record-set already exists

Reply:
Renamed it